### PR TITLE
Fixing codellb dap 

### PR DESCRIPTION
**Pull request by Freyskeyd**
Hello,

In order to fix the dap settings to allow debugging we discuss of solution in the issue #157 

I came across some more issue trying to fix it but I found a way to prevent some of the concurrency issue.
The solution is mostly a cherrypick of the @simrat39 code in the issue but with an asynchronous schedule of the server to prevent concurrency issue.
For now I add 500ms delay because I've not found any better solution. I'm open to suggestion on this part.

 
Signed-off-by: Freyskeyd <simon.paitrault@gmail.com>

### CODE REVIEW

**Comment by simrat39** [comment]
there should be a better way than delaying by an arbitrary amount, will have to look at the vscode extension's source code to figure that out tbh

**Comment by Freyskeyd** [comment]
@simrat39 I checked the code of [novsc adapter](https://github.com/vadimcn/vscode-lldb/blob/54d4fbb2dead83e98bd44d07057b67efc4305531/extension/novsc/adapter.ts#L39) but I can't find a reason why it is working. I'm not really used to typescript nor vscode code base tho.

Does switching from LibUV to plenary.nvim can help ?

**Comment by Suyashtnt** [comment]
After some testing this fix does indeed work

File: lua/rust-tools/dap.lua [hunk 1]
```diff
@@ -64,6 +60,16 @@ function M.get_codelldb_adapter(codelldb_path, liblldb_path)
         end)
       end
     end)
+
+    vim.defer_fn(function()
+      vim.schedule(function()
```
Comment (againxx) [inline lua/rust-tools/dap.lua:65]: `defer_fn` automatically wraps the function in `vim.schedule`, maybe
this `vim.schedule` can be removed

Comment (againxx) [inline lua/rust-tools/dap.lua:65]: `defer_fn` automatically wraps the function in `vim.schedule`, maybe
this `vim.schedule` can be removed

**Comment by yi-ge** [comment]
Thank you very much!

**Comment by simrat39** [comment]
Looks like dap added support upstream, so we no longer need hacks like these. I pushed a fix to the https://github.com/simrat39/rust-tools.nvim/tree/modularize_and_inlay_rewrite branch. Please let me know how it works so we can close this

**Comment by LC0117** [comment]
@simrat39 I have tried the new branch, and the `RustDebuggables` option works fine using codelldb.

**Comment by simrat39** [comment]
New branch with upstream fix merged. Closing.

**Pull request closed without merging**